google/auth/credentials.py

```diff
@@ -24,6 +24,7 @@
 from google.auth import metrics
 from google.auth._refresh_worker import RefreshThreadManager
 
+DEFAULT_TRUST_BOUNDARY = {"locations": [], "encoded_locations": "0x0"}
```

Comment:
I would not use a global dictionary like this, it is prone to race conditions. 

It would be shared by every credential object instantiated with it.

Reply:
How about set the constant itself as a deepcopy?